resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Constants.scala

```diff
@@ -86,6 +86,7 @@ private[spark] object Constants {
   val DEFAULT_EXECUTOR_CONTAINER_NAME = "spark-kubernetes-executor"
   val MEMORY_OVERHEAD_MIN_MIB = 384L
   val NON_JVM_MEMORY_OVERHEAD_FACTOR = 0.4d
+  val KUBERNETES_MAX_NAME_LENGTH = 63
```

Comment:
Also, could you remove the following at line 80?
```scala
val POD_TEMPLATE_CONFIGMAP = "podspec-configmap"
```

Reply:
In my first version of the PR it was unused but I changed it back to use this constant.
Unless we want to have it more like `DriverServiceFeatureStep` where the suffix is stored in the companion object.

Reply:
By the way @stijndehaes I haven't been able to find the document that says the max name length of any configmap is 63 char long.  From my quick command line test, I found it can be well over beyond that number.  Could you confirm this limit is needed. Well, I do agree it is always good to be defensive but I found there is no such shortening logic for driver configmap either. So, could you confirm? Thanks.

Reply:
I'd vote for using the Constant.scala for store the value.

Reply:
@yuj for me it is fine using the constant. Let's not change more than needed :)

Reply:
@stijndehaes right.